Tests/SwiftSyntaxTest/SyntaxComparisonTests.swift

```diff
@@ -0,0 +1,178 @@
+
+    let matcher = try SubtreeMatcher("struct A { func f() { } }")
+    try expectations(matcher.findFirstDifference(baseline: expected))
+  }
+
+  public func testAdditionalNode() throws {
+    let expected = Syntax(makeFunc(identifier: SyntaxFactory.makeIdentifier("f")))
+    func expectations(_ diff: TreeDifference?, file: StaticString = #filePath, line: UInt = #line) throws {
+      let diff = try XCTUnwrap(diff, file: file, line: line)
+      XCTAssertEqual(diff.reason, .additionalNode)
+    }
+
+    let actual = Syntax(makeFunc(identifier: SyntaxFactory.makeIdentifier("f"), body: makeBody(statementCount: 1)))
+    try expectations(actual.findFirstDifference(baseline: expected))
+
+    let matcher = try SubtreeMatcher("""
+      struct A {
+        func f() {
+          0
+        }
+      }
+      """)
+    try expectations(matcher.findFirstDifference(baseline: expected))
+  }
+
+  public func testMultipleSubtreeMatches() throws {
+    let expectedFunc = Syntax(makeFunc(identifier: SyntaxFactory.makeIdentifier("f")))
+    let expectedBody = Syntax(makeBody())
+
+    let matcher = try SubtreeMatcher("""
+      struct A {
+        #^FUNC^#
+        func f() #^BODY^#{
+          0
+        }
+      }
+      """)
+    let funcDiff = try XCTUnwrap(matcher.findFirstDifference(afterMarker: "FUNC", baseline: expectedFunc))
+    XCTAssertEqual(funcDiff.reason, .additionalNode)
+
+    let bodyDiff = try XCTUnwrap(matcher.findFirstDifference(afterMarker: "BODY", baseline: expectedBody))
+    XCTAssertEqual(bodyDiff.reason, .additionalNode)
+  }
+
+  private func makeFunc(identifier: TokenSyntax, keyword: TokenSyntax = SyntaxFactory.makeFuncKeyword(),
+                        body: CodeBlockSyntax? = nil, indent: Int = 0) -> FunctionDeclSyntax {
+    let funcBody: CodeBlockSyntax
+    if let body {
+      funcBody = body
+    } else {
+      funcBody = makeBody()
+    }
+    let emptySignature = SyntaxFactory.makeFunctionSignature(input: SyntaxFactory.makeParameterClause(leftParen: SyntaxFactory.makeLeftParenToken(),
+                                                                                                      parameterList: SyntaxFactory.makeFunctionParameterList([]),
+                                                                                                      rightParen: SyntaxFactory.makeRightParenToken()),
+                                                             asyncOrReasyncKeyword: nil, throwsOrRethrowsKeyword: nil, output: nil)
+    let fd = SyntaxFactory.makeFunctionDecl(attributes: nil, modifiers: nil,
+                                            funcKeyword: keyword, identifier: identifier, genericParameterClause: nil,
+                                            signature: emptySignature, genericWhereClause: nil, body: funcBody)
+    if indent > 0 {
+      return fd.withLeadingTrivia(.spaces(indent))
+    }
+    return fd
+  }
+
+  private func makeBody(statementCount: Int = 0) -> CodeBlockSyntax {
```

Comment:
Suggested doc comment.

````suggestion
  /// Creates a `CodeBlockSyntax` that consists of `statementCount` integer literals with increasing values. I.e. `makeBody(statementCount: 2)` generates
  ```
  {
    0
    1
  }
  ```
  private func makeBody(statementCount: Int = 0) -> CodeBlockSyntax {
````